clang/include/clang/Frontend/CompileJobCacheResult.h

```diff
@@ -46,11 +46,21 @@ class CompileJobCacheResult : public ObjectProxy {
   llvm::Error
   forEachOutput(llvm::function_ref<llvm::Error(Output)> Callback) const;
 
+  /// Loads all outputs concurrently and passes the resulting \c ObjectProxy
+  /// objects to \p Callback. If there was an error during loading then the
+  /// callback will not be invoked.
+  llvm::Error forEachLoadedOutput(
```

Comment:
Why is this a separate API from `forEachOutput`? It's not clear to me why this isn't just a new implementation for that function.

Reply:
A caller may want to just get all the output IDs without loading anything, in which case they'd use `forEachOutput`.

Reply:
Oh sorry, I didn't realize `Output` contained a Ref instead of a Proxy.  SGTM.